security/src/main/scala/pl/touk/nussknacker/ui/security/api/ImpersonationContext.scala

```diff
@@ -0,0 +1,25 @@
+package pl.touk.nussknacker.ui.security.api
+
+import com.typesafe.config.Config
+import sttp.client3.SttpBackend
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait ImpersonationContext {
```

Comment:
I'm not sure if we shouldn't do it in a little bit different way. 

Let's assume that we have these two things: `AuthenticationManager` and `AuthenticationResources`. The `AuthenticationManager` is just a wrapper or adapter for the `AuthenticationResources` class. It's fine for now. 

The `AuthenticationResources` interface shows us that it's responsible for providing authentication methods (implementations). For me, it'd be nice, if it could say more: 
* I'm responsible for the impersonation support
* I'm responsible for the anonymous access support

It could look like that:
```scala
trait AuthenticationResources extends ... {
  this: ImpersonationSupport with AnonymousAccessSupport
}
```

and the `ImpersonationSupport` trait could look like this:
```scala
trait ImpersonationSupport {

  def getImpersonatedUserData(impersonatedUserIdentity: String): Option[ImpersonatedUserData]

}

trait NoImpersonationSupport extends {
   protected def getImpersonatedUserData(impersonatedUserIdentity: String): Option[ImpersonatedUserData] = None 
}
```

it seems that the things from ImpersonatedAccess we could move to AuthenticationManager.

The same could be done in case of the anonymous access.

Then the e.g. `OAuth2AuthenticationResources` implementation could look like this:
```scala
class OAuth2AuthenticationResources(...)
  extends AuthenticationResources
   with NoImpersonationSupport
   with AnonymousAccessSupport { ... }
```
WDYT about this appoarch? TBH it's rather a little modification compared to the current solution.